sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_data_index/data_index.py

```diff
@@ -0,0 +1,163 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument
+
+from marshmallow import fields, post_load
+
+from azure.ai.ml._schema.assets.data import DataSchema
+from azure.ai.ml._schema.core.fields import NestedField, StringTransformedEnum, UnionField
+from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
+from azure.ai.ml._schema.job.input_output_entry import DataInputSchema, MLTableInputSchema, generate_path_property
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants._common import AzureMLResourceType
+
+
+class DataIndexTypes:
+    """DataIndexTypes is an enumeration of values for the types out indexes which can be written to by DataIndex."""
+
+    ACS = "acs"
+    """Azure Cognitive Search index type."""
+    FAISS = "faiss"
+    """Faiss index type."""
+
+
+class CitationRegexSchema(metaclass=PatchedSchemaMeta):
+    match_pattern = fields.Str(
+        required=True,
+        metadata={"description": "Regex to match citation in the citation_url + input file path. e.g. '\\1/\\2'"},
+    )
+    replacement_pattern = fields.Str(
+        required=True,
+        metadata={"description": "Replacement string for citation. e.g. '(.*)/articles/(.*)'"},
```

Comment:
Probably want to be consistent with Examples + Component descriptions
'(.*)/articles/(.*)(\\.[^.]+)$'